include/clang/AST/ASTImporter.h

```diff
@@ -56,12 +57,19 @@ class TypeSourceInfo;
   /// Imports selected nodes from one AST context into another context,
   /// merging AST nodes where appropriate.
   class ASTImporter {
+    friend class ASTNodeImporter;
   public:
     using NonEquivalentDeclSet = llvm::DenseSet<std::pair<Decl *, Decl *>>;
     using ImportedCXXBaseSpecifierMap =
         llvm::DenseMap<const CXXBaseSpecifier *, CXXBaseSpecifier *>;
 
   private:
+
+    /// Pointer to the import specific lookup table, which may be shared
+    /// amongst several ASTImporter objects.
+    /// If not set then the original C/C++ lookup is used.
+    ASTImporterLookupTable *LookupTable = nullptr;
```

Comment:
Can we not use a shared pointer for this?

Reply:
In the current way of usage shared pointer is not needed. But a notice can be here that the lookup table is an externally managed resource (and should exist during the lifetime of the ASTImporter object). This variable can be an optional too (if it is possible that it contains a reference to ensure non-null value).